internal/users/locking/testutils.go

```diff
@@ -17,28 +35,163 @@ var overrideLocked atomic.Bool
 func Z_ForTests_OverrideLocking() {
 	testsdetection.MustBeTesting()
 
+	overriddenMu.Lock()
+	defer overriddenMu.Unlock()
+
+	overriddenWriteLockImpl = append(overriddenWriteLockImpl, writeLockImpl)
 	writeLockImpl = func() error {
 		if !overrideLocked.CompareAndSwap(false, true) {
-			return ErrLock
+			return fmt.Errorf("%w: already locked!", ErrLock)
+		}
+
+		log.Debug(context.Background(), "TestOverride: Local entries locked!")
+		return nil
+	}
+
+	overriddenWriteUnlockImpl = append(overriddenWriteUnlockImpl, writeUnlockImpl)
+	writeUnlockImpl = func() error {
+		if !overrideLocked.CompareAndSwap(true, false) {
+			return fmt.Errorf("%w: already unlocked!", ErrUnlock)
+		}
+
+		log.Debug(context.Background(), "TestOverride: Local entries unlocked!")
+		return nil
+	}
+}
+
+// Z_ForTests_OverrideLockingWithCleanup is a function to override the locking
+// functions for testing purposes.
+// It simulates the real behavior but without actual file locking.
+// This implicitly calls [Z_ForTests_RestoreLocking] once the test is completed.
+//
+// nolint:revive,nolintlint // We want to use underscores in the function name here.
+func Z_ForTests_OverrideLockingWithCleanup(t *testing.T) {
+	t.Helper()
+
+	testsdetection.MustBeTesting()
+
+	Z_ForTests_OverrideLocking()
+	t.Cleanup(Z_ForTests_RestoreLocking)
+}
+
+// Z_ForTests_OverrideLockingAsLockedExternally is a function to
+// override the locking functions for testing purposes.
+// It simulates the real behavior but without actual file locking.
+// This locks the database when called, the lock can be both removed using
+// [WriteUnlock] or cancelling the context. It also implicitly calls
+// [Z_ForTests_RestoreLocking] once the test is completed.
+//
+// nolint:revive,nolintlint // We want to use underscores in the function name here.
+func Z_ForTests_OverrideLockingAsLockedExternally(t *testing.T, ctx context.Context) {
```

Comment:
It's not clear to me what this function does "as externally", in contrast to `Z_ForTests_OverrideLocking`.

Reply:
ok, thanks for explaining! let's try to make the purpose a bit clearer in the comment. how about this:

```
// Z_ForTests_OverrideLockingAsLockedExternally simulates a scenario where the
// user database is locked by an external process.
//
// When called, it marks the user database as locked, causing any subsequent
// locking attempts by authd (via WriteRecLock) to block until the provided
// context is cancelled.
//
// This does not use real file locking. The lock can be released either
// by cancelling the context or by calling [WriteUnlock]. After the test,
// [Z_ForTests_RestoreLocking] is called automatically to restore normal behavior.
```